timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -308,6 +298,103 @@ void declare_context(py::module &m) {
+
+                if (reference_idx < 0 || reference_idx >= N) {
+                    throw std::runtime_error("reference idx must be at least 0 and less than " + std::to_string(N));
+                }
+                std::vector<int> vec_selection_idxs(selection_idxs.size());
+                std::memcpy(vec_selection_idxs.data(), selection_idxs.data(), vec_selection_idxs.size() * sizeof(int));
+                verify_atom_idxs(N, vec_selection_idxs);
+                std::set<int> selection_set(vec_selection_idxs.begin(), vec_selection_idxs.end());
+                if (selection_set.find(reference_idx) != selection_set.end()) {
+                    throw std::runtime_error("reference idx must not be in selection idxs");
+                }
+
+                std::array<std::vector<double>, 2> result = ctxt.multiple_steps_local_selection(
+                    n_steps, reference_idx, vec_selection_idxs, burn_in, x_interval, radius, k);
+                const int D = 3;
+                const int F = result[0].size() / (N * D);
+                py::array_t<double, py::array::c_style> out_x_buffer({F, N, D});
+                std::memcpy(out_x_buffer.mutable_data(), result[0].data(), result[0].size() * sizeof(double));
+
+                py::array_t<double, py::array::c_style> box_buffer({F, D, D});
+                std::memcpy(box_buffer.mutable_data(), result[1].data(), result[1].size() * sizeof(double));
+                return py::make_tuple(out_x_buffer, box_buffer);
+            },
+            py::arg("n_steps"),
+            py::arg("reference_idx"),
+            py::arg("selection_idxs"),
+            py::arg("burn_in") = 500, // This is arbitrarily selected as a default, TODO make informed choice
+            py::arg("store_x_interval") = 0,
+            py::arg("radius") = 1.2,
+            py::arg("k") = 10000.0,
+            R"pbdoc(
+        Take multiple steps using a selection of free particles restrained to a reference particle. Useful for avoiding the bias
+        introduced by switching on and off the restraint on different particles as is done with multiple_steps_local.
+
+        Running a barostat and local MD at the same time are not currently supported. If a barostat is
+        assigned to the context, the barostat won't run.
+
+        Note: Running this multiple times with small number of steps (< 100) may result in a vacuum around the local idxs due to
+        discretization error caused by switching on the restraint after a particle has moved beyond the radius.
+
+        F = iterations / store_x_interval
+
+        The first call to `multiple_steps_local_selection` takes longer than subsequent calls, if initialize_local_md has not been called previously,
+        initializes potentials needed for local md.
+
+        Parameters
+        ----------
+        n_steps: int
+            Number of steps to run.
+
+        selection_idxs: np.array of int32
```

Comment:
```suggestion
        reference_idxs, selection_idxs: int, np.array of int32
```

Reply:
Fixed in [e5f18c6](https://github.com/proteneer/timemachine/pull/1005/commits/e5f18c6d278dd64575877e9a8212a502847f91f3)